packages/block-library/src/table/edit.js

```diff
@@ -414,31 +414,35 @@ function TableEdit( {
 							},
 							columnIndex
 						) => (
-							<RichText
-								tagName={ CellTag }
-								key={ columnIndex }
-								className={ classnames(
-									{
-										[ `has-text-align-${ align }` ]: align,
-									},
-									'wp-block-table__cell-content'
-								) }
-								scope={ CellTag === 'th' ? scope : undefined }
-								colSpan={ colspan }
-								rowSpan={ rowspan }
-								value={ content }
-								onChange={ onChange }
-								onFocus={ () => {
-									setSelectedCell( {
-										sectionName: name,
-										rowIndex,
-										columnIndex,
-										type: 'cell',
-									} );
-								} }
-								aria-label={ cellAriaLabel[ name ] }
-								placeholder={ placeholder[ name ] }
-							/>
+							<CellTag key={ columnIndex }>
+								<RichText
```

Comment:
This will likely break visuals but it works due to the cell itself remaining a table cell. The editable content is now done within the cell. This will also likely cause a slightly different back-end vs. the front-end but it isn't up for debate. It simply does not work when you convert a table cell to an editable field. The entire table loses semantic meaning with screen readers.

Reply:
Why do we need role=textbox if it breaks things? Isn't contenteditable enough?

Reply:
@ellatrix With my changes, it comes with 2 bonuses.
1. The table semantics are respected.
2. The editable areas inside the table cells have better text input semantics.

All in all, a big win.